components/property_info.rst

```diff
@@ -0,0 +1,295 @@
+
+* :doc:`Install it via Composer </components/using_components>` (``symfony/property-info``
+  on `Packagist`_);
+* Use the official Git repository (https://github.com/symfony/PropertyInfo).
+
+.. include:: /components/require_autoload.rst.inc
+
+To use the :class:`Symfony\\Component\\PropertyInfo\\Extractor\\PhpDocExtractor`,
+install `phpDocumentator Reflection`_.
+To use the :class:`Symfony\\Component\\PropertyInfo\\Extractor\\SerializerExtractor`
+extractor, install the :doc:`Serializer component </components/serializer>`.
+To use the :class:`Symfony\\Bridge\\Doctrine\\PropertyInfo\\DoctrineExtractor`,
+install the Doctrine Bridge and the `Doctrine ORM`_.
+
+Usage
+-----
+
+Using the PropertyInfo component is straightforward. You need to register
+all metadata extractors you want to use in the constructor of the :class:`Symfony\\Component\\PropertyInfo\\PropertyInfo`
+class::
+
+    use Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor;
+    use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+    use Symfony\Component\PropertyInfo\PropertyInfo;
+
+    $reflectionExtractor = new ReflectionExtractor();
+    $phpDocExtractor = new PhpDocExtractor();
+
+    $propertyInfo = new PropertyInfo(
+        array($reflectionExtractor),
+        array($phpDocExtractor, $reflectionExtractor),
+        array($phpDocExtractor),
+        array($reflectionExtractor)
+    );
+
+An array of implementations of :class:`Symfony\\Component\\PropertyInfo\\PropertyListExtractorInterface`
+must be passed as first parameter. These extractors are responsible of extracting
+the list of properties of a class.
+
+An array of implementations of :class:`Symfony\\Component\\PropertyInfo\\PropertyTypeExtractorInterface`
+must be passed as second parameter. These extractors are responsible of extracting
+types of a property.
+
+An array of implementations of :class:`Symfony\\Component\\PropertyInfo\\PropertyDescriptionExtractorInterface`
+must be passed as third parameter. These extractors are responsible of extracting
+short and long DocBlock description of a property.
+
+Finally, an array of implementations of :class:`Symfony\\Component\\PropertyInfo\\PropertyAccessExtractorInterface`
+must be passed as fourth parameter. These extractors are responsible of guessing
+if a property is readable or writable.
+
+The order of registration matter: the data returned will be the one returned
+by the first extractor if different than ``null``.
+
+
+Once instantiated, the ``PropertyInfo`` class can be used to retrieve info
+about properties of a class::
+
+    class MyClass
+    {
+        /**
+         * The short description of foo.
+         *
+         * And here is its extended description.
+         *
+         * @var string
+         */
+        public $foo;
+
+        /**
+         * Virtual property.
+         */
+        private function setBar(array $bar)
+        {
+        }
+    }
+
+    var_dump($propertyInfo->getProperties('MyClass'));
```

Comment:
to advocate the Symfony VarDumper component a bit, we always use `dump` in the Symfony docs. I also think it's easier to follow if you put the output directly after each line:
```php
dump($propertyInfo->getProperties('MyClass'));
/* Produces: array(0 => 'foo') */
...
```